test/test_verifylib.py

```diff
@@ -0,0 +1,502 @@
+#!/usr/bin/env python
+
+"""
+<Program Name>
+  test_verifylib.py
+
+<Author>
+  Lukas Puehringer <lukas.puehringer@nyu.edu>
+
+<Started>
+  Nov 07, 2016
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Test verifylib functions.
+
+"""
+
+import unittest
+from toto.models.link import Link
+from toto.models.layout import Step, Inspection
+from toto.verifylib import verify_delete_rule, verify_create_rule, \
+    verify_match_rule, verify_item_rules, verify_all_item_rules
+from toto.exceptions import RuleVerficationFailed
+
+class TestVerifyDeleteRule(unittest.TestCase):
+  """Test verifylib.verify_delete_rule(rule, artifact_queue) """
+
+  def setUp(self):
+    """ Setup artifact queues. """
+    self.artifact_queue = ["foo"]
+    self.artifact_queue_empty = []
+
+
+  def test_fail_delete_file(self):
+    """["DELETE", "foo"], matches foo (not deleted), fails. """
+
+    rule = ["DELETE", "foo"]
+    with self.assertRaises(RuleVerficationFailed):
+      verify_delete_rule(rule, self.artifact_queue)
+
+
+  def test_fail_delete_star(self):
+    """["DELETE", "*"], matches * in non-empty queue (not deleted), fails. """
+
+    rule = ["DELETE", "*"]
+    with self.assertRaises(RuleVerficationFailed):
+        verify_delete_rule(rule, self.artifact_queue)
+
+
+  def test_pass_delete_file(self):
+    """["DELETE", "bar"] does not match bar (deleted), passes. """
+
+    rule = ["DELETE", "bar"]
+    self.assertIsNone(
+        verify_delete_rule(rule, self.artifact_queue))
+
+
+  def test_pass_delete_star(self):
+    """["DELETE", "*"], does not match * in empty queue (deleted), passes. """
+
+    rule = ["DELETE", "*"]
+    self.assertIsNone(
+        verify_delete_rule(rule, self.artifact_queue_empty))
+
+
+  def test_pass_ignore_case_keyword(self):
+    """["delete", "bar"], ["DELETE", "bar"], ignores keyword case, passes. """
+
+    rule1 = ["delete", "bar"]
+    rule2 = ["DELETE", "bar"]
+    self.assertIsNone(
+        verify_delete_rule(rule1, self.artifact_queue))
+    self.assertIsNone(
+        verify_delete_rule(rule2, self.artifact_queue))
+
```

Comment:
This is actually the first time that I adhere to the [blank line style guide line](https://github.com/secure-systems-lab/code-style-guidelines#blank-lines).  :)
So far I didn't use this particular guide line because I thought it unnecessarily spreads up the code.
But in this case - many very short functions with one liner doc strings - I think it makes the code look less cluttered.

Anyway, you are probably right that we should do it either everywhere or nowhere.